shared/test/test_state_abbr.rb

```diff
@@ -59,6 +59,61 @@ def test_get_us_state_from_abbr_washington_dc
     assert_equal 'Washington DC', get_us_state_from_abbr('DC', true)
   end
 
+  def test_get_us_state_abbr_from_name_lowercase_symbol
+    assert_equal 'IL', get_us_state_abbr_from_name(:illinois)
+    assert_equal 'WA', get_us_state_abbr_from_name(:washington)
+  end
+
+  def test_get_us_state_abbr_from_name_uppercase_symbol
+    assert_equal 'IL', get_us_state_abbr_from_name(:ILLINOIS)
+    assert_equal 'WA', get_us_state_abbr_from_name(:WASHINGTON)
+  end
+
+  def test_get_us_state_abbr_from_name_lowercase_string
+    assert_equal 'IL', get_us_state_abbr_from_name('illinois')
+    assert_equal 'WA', get_us_state_abbr_from_name('washington')
+  end
+
+  def test_get_us_state_abbr_from_name_uppercase_string
+    assert_equal 'IL', get_us_state_abbr_from_name('ILLINOIS')
+    assert_equal 'WA', get_us_state_abbr_from_name('WASHINGTON')
+  end
+
+  def test_get_us_state_abbr_from_name_string_with_whitespace
+    assert_equal 'IL', get_us_state_abbr_from_name(' Illinois   ')
+    assert_equal 'WA', get_us_state_abbr_from_name('  Washington  ')
+  end
+
+  def test_get_us_state_abbr_from_name_nonstate
+    assert_nil get_us_state_abbr_from_name('Nonexistent')
+    assert_nil get_us_state_abbr_from_name('Nonexistent', false)
+    assert_nil get_us_state_abbr_from_name('Nonexistent', true)
+  end
+
+  def test_get_us_state_abbr_from_name_with_include_dc
+    assert_equal 'IL', get_us_state_abbr_from_name('Illinois', false)
+    assert_equal 'IL', get_us_state_abbr_from_name('Illinois', true)
+    assert_equal 'WA', get_us_state_abbr_from_name('Washington', false)
+    assert_equal 'WA', get_us_state_abbr_from_name('Washington', true)
+  end
+
+  def test_get_us_state_abbr_from_name_washington_dc
+    assert_nil get_us_state_abbr_from_name(:washington_dc)
```

Comment:
You can also DRY this with nested loops - up to you if you want to do it

Reply:
I stuck with the existing pattern from the rest of this test, and I'd rather not invest more time there now